controllers/cluster_controller.go

```diff
@@ -665,8 +665,7 @@ func (r *ClusterReconciler) ensureHealthyPVCsAnnotation(
 	for _, pvcName := range cluster.Status.HealthyPVC {
 		pvc := resources.getPVC(pvcName)
 		if pvc == nil {
-			contextLogger.Warning("unable to find pvc to annotate it as ready", "pvc", pvc.Name)
-			continue
+			return fmt.Errorf("pvc not found")
```

Comment:
I'm not fully aware of the context behind this, but I agree with you, this seems a programmatic error.
We should raise an error here and perhaps improve the error message including the name of the PVC that we don't found.